src/auto-evo/GenerationRecord.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace AutoEvo
+{
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    ///   Record of Auto-Evo results and species data for a given generation.
+    /// </summary>
+    [JsonObject(IsReference = true)]
+    [UseThriveSerializer]
+    public class GenerationRecord
+    {
+        /// <summary>
+        ///   Player generation in the game (starting at zero).
+        /// </summary>
+        [JsonProperty]
+        public int Generation;
+
+        /// <summary>
+        ///   Total in-game time elapsed since its beginning.
+        /// </summary>
+        [JsonProperty]
+        public double TimeElapsed;
+
+        /// <summary>
+        ///   Auto-Evo results for this generation.
+        /// </summary>
+        [JsonProperty]
+        public RunResults AutoEvoResult = null!;
```

Comment:
In this case as this is a normal C# class I'd prefer to have a constructor that takes in the AutoEvoResults and AllSpecies, that way the code doesn't need null suppression and is much more likely to be used correctly without causing null bugs.

You can see the Patch and region and the map classes on how I handled the JSON constructors etc. there.